common/scala/src/main/scala/whisk/core/database/memory/MemoryAttachmentStore.scala

```diff
@@ -0,0 +1,185 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.database.memory
+
+import akka.actor.ActorSystem
+import akka.event.Logging.ErrorLevel
+import akka.http.scaladsl.model.ContentType
+import akka.stream.{ActorMaterializer, OverflowStrategy, QueueOfferResult}
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.util.{ByteString, ByteStringBuilder}
+import whisk.common.{Logging, StartMarker, TransactionId}
+import whisk.common.LoggingMarkers.DATABASE_ATT_GET
+import whisk.common.LoggingMarkers.DATABASE_ATT_SAVE
+import whisk.common.LoggingMarkers.DATABASE_ATT_DELETE
+import whisk.core.database._
+import whisk.core.entity.{DocId, DocInfo}
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.reflect.ClassTag
+
+object MemoryAttachmentStoreProvider extends AttachmentStoreProvider {
+  override def makeStore[D <: DocumentSerializer: ClassTag]()(implicit actorSystem: ActorSystem,
+                                                              logging: Logging,
+                                                              materializer: ActorMaterializer): AttachmentStore =
+    new MemoryAttachmentStore(implicitly[ClassTag[D]].runtimeClass.getSimpleName.toLowerCase)
+}
+
+class MemoryAttachmentStore(dbName: String)(implicit system: ActorSystem,
```

Comment:
Can you add a comment describing this store, what it's supposed to do, how it does it, etc.

Reply:
Done